### ISSUE TITLE
Add filename to "Filesystem config update failure"  warning

### ISSUE THREAD

**Issue by MichaelSasser** [envoyproxy/envoy#24218]
*Title*: *Misconfigured file is hard to find on "Filesystem config update failure"*

*Description*:

When I deploy a new service to the testing environment and I misconfigured something, I find a line like the following in the logs.
```log
[2022-11-27 04:51:34.312][1][warning][config] [source/common/config/filesystem_subscription_impl.cc:98] Filesystem config update failure: yaml-cpp: error at line 5, column 5: end of sequence not found
```
The line tells me, I made a mistake and where in the file the mistake happened, but doesn't describe which file has issues.

If the warning was changed to something like the following, it wouldn't be so frustrating.

```log
[2022-11-27 04:51:34.312][1][warning][config] [source/common/config/filesystem_subscription_impl.cc:98] Filesystem config update failure: yaml-cpp: error in /foo/bar/rds.yaml at line 5, column 5: end of sequence not found
```
(In my example I've added `in /foo/bar/rds.yaml`)

**Comment by wbpcode** [comment]
Sounds reasonable requirement.

**Comment by MichaelSasser** [comment]
I think the issue can be closed as #24503 was merged.
